subsys/bluetooth/host/id.c

```diff
@@ -762,7 +781,8 @@ void bt_id_add(struct bt_keys *keys)
 		}
 	}
 
-	if (bt_dev.le.rl_entries == bt_dev.le.rl_size) {
+	key_already_added = (bt_keys_find_addr(keys->id, &keys->addr) != NULL);
```

Comment:
Regarding:
```
isn't this always going to be true?

```

Not the first time a device bonds with the Zephyr device since the IA in keys->addr would not be in key_pool. But it will be true for all subsequent bondings from that device.

Also, I think it would be safer to check the key_pool for the IA, instead of using the BT_KEYS_ID_ADDED flag, since the key_pool either actually has the IA or not, and the BT_KEYS_ID_ADDED flag should, of course, be synchronized with this, but checking the key_pool directly is safer and more robust. What do you think?

Regarding:
```
also, since we can "update" entry now, every
bt_dev.le.rl_entries++;
keys->state |= BT_KEYS_ID_ADDED;

should be done only if BT_KEYS_ID_ADDED is not set yet (to make sure bt_dev.le.rl_entries is correct)
```

BT_KEYS_ID_ADDED not set is be the same as key_already_added == false, and

```c
bt_dev.le.rl_entries++;
keys->state |= BT_KEYS_ID_ADDED;
```

Is not executed if key_already_added == false

Also, it is only executed if switching to host side resolving or hci_id_add is successful. This is what you want to ensure right? Is this not ok then?